word/views.py

```diff
@@ -41,28 +41,56 @@ def get_context_data(self, **kwargs):
         return context
 
     def post(self, request, *args, **kwargs):
+        uploaded_file_url = None
+        if request.method == "POST" and "myfile" in request.FILES:
+            myfile = request.FILES["myfile"]
+            fs = FileSystemStorage()
+            filename = fs.save(myfile.name, myfile)
+            if filename.endswith(".mp3") or filename.endswith(".wma") or\
+                filename.endswith(".wav") or filename.endswith(".m4a"):
+                    uploaded_file_url = fs.url(filename)
+            else:
+                filename = None
+                uploaded_file_url = fs.url(filename)
+
         word = request.POST.get('word')
+        tags = request.POST.getlist('tags')
         ipa = request.POST.get('ipa')
-        version = request.POST.get('version')
+        version_id = request.POST.get('version')
         location = request.POST.get('location')
         description_short = request.POST.get('desc_short')
         description_long = request.POST.get('desc_long')
+        synonyms = request.POST.getlist('synonyms')
+        wiktionary_link = request.POST.get('wiktionary_link')
 
-        version = WordVersion.objects.get(pk=version)
-
+        version_object = WordVersion.objects.get(pk=version_id)
         desc = Description.objects.create(
             short=description_short,
             extended=description_long,
-            language=version.language,
+            language=version_object.language,
         )
+
         word = Word.objects.create(
             word=word,
             ipa=ipa,
             status='SUG',
-            version=version,
+            version=version_object,
+            audio=uploaded_file_url,
             submitter=request.user.profile,
+            wiktionary_link = wiktionary_link,
         )
         word.desc.add(desc)
+
+        for tag in tags:
+            tag_object, _ = Tag.objects.get_or_create(name=tag.lower())
+            word.tags.add(tag_object)
+        print("=====================", tags)

```

Comment:
Is this print debug code? Otherwise it is better to use the logging module.

Reply:
yes. I am not familiar with that but I will try.